tests/e2e/data/observatory/catalog/test_download.py

```diff
@@ -125,3 +128,70 @@ def test_geography_to_dataframe_private(self):
         expected_df = pandas.read_csv(file_path('files/private-geography.csv'))
 
         assert df.equals(expected_df)
+
+    def test_dataset_to_csv_public_with_sql_query(self):
+        sql_query = 'select * from {dataset} order by geoid limit 2'
+        public_dataset.to_csv(self.tmp_file, self.credentials, sql_query=sql_query)
+
+        assert os.path.isfile(self.tmp_file)
+
+        df = pandas.read_csv(self.tmp_file)
+        expected_df = pandas.read_csv(file_path('files/public-dataset-ordered-and-limited.csv'))
+
+        assert df.equals(expected_df)
+
+    def test_dataset_to_csv_public_with_sql_query_and_add_geom(self):
+        sql_query = 'select * from {dataset} order by geoid limit 2'
+        add_geom = True
+        public_dataset.to_csv(self.tmp_file, self.credentials, sql_query=sql_query, add_geom=add_geom)
+
+        assert os.path.isfile(self.tmp_file)
+
+        df = pandas.read_csv(self.tmp_file)
+        expected_df = pandas.read_csv(file_path('files/public-dataset-join-geography.csv'))
+
+        assert df.equals(expected_df)
+
+    def test_geography_to_csv_public_with_sql_query(self):
+        sql_query = 'select * from {geography} order by geoid limit 2'
+        public_geography.to_csv(self.tmp_file, self.credentials, sql_query=sql_query)
+
+        assert os.path.isfile(self.tmp_file)
+
+        df = pandas.read_csv(self.tmp_file)
+        expected_df = pandas.read_csv(file_path('files/public-geography.csv'))
+
+        assert df.equals(expected_df)
+
+    def test_dataset_to_dataframe_public_with_sql_query(self):
+        sql_query = 'select * from {dataset} order by geoid limit 2'
+        df = public_dataset.to_dataframe(self.credentials, sql_query=sql_query)
+        df.to_csv(self.tmp_file, index=False)
+
+        df = pandas.read_csv(self.tmp_file)
+        expected_df = pandas.read_csv(file_path('files/public-dataset-ordered-and-limited.csv'))
+
+        assert df.equals(expected_df)
+
+    @pytest.mark.skip()  # TODO implement equals check using a tolerance
+    def test_dataset_to_dataframe_public_with_sql_query_and_add_geom(self):
+        sql_query = 'select * from {dataset} order by geoid limit 2'
+        add_geom = True
+        df = public_dataset.to_dataframe(self.credentials, sql_query=sql_query, add_geom=add_geom)
+        df.to_csv(self.tmp_file, index=False)
```

Comment:
why do you need to store `df` into a file for then reading and comparing with the `expected_df`?

Reply:
Just followed what it's done in the rest of the test.